src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java

```diff
@@ -269,20 +277,50 @@ private ContextMenu createFileMenu(BibEntryTableViewModel entry, List<LinkedFile
     }
 
     /**
-     * Creates a column which shows an icon instead of the textual content
+     * Creates a column for DOIs or URLs.
+     * The {@code firstField} is preferred to be shown over {@code secondField}.
      */
-    private TableColumn<BibEntryTableViewModel, String> createIconColumn(JabRefIcon icon, String firstField, String secondField) {
+    private TableColumn<BibEntryTableViewModel, String> createUrlOrDoiColumn(JabRefIcon icon, String firstField, String secondField) {
         TableColumn<BibEntryTableViewModel, String> column = new TableColumn<>();
         column.setGraphic(icon.getGraphicNode());
         column.getStyleClass().add(ICON_COLUMN);
         setExactWidth(column, GUIGlobals.WIDTH_ICON_COL);
-        column.setCellValueFactory(cellData -> EasyBind.monadic(cellData.getValue().getField(firstField)).orElse(cellData.getValue().getField(secondField)));
+        column.setCellValueFactory(cellData -> EasyBind.monadic(cellData.getValue().getField(firstField)).map(x -> firstField).orElse(EasyBind.monadic(cellData.getValue().getField(secondField)).map(x -> secondField)));
```

Comment:
whats the purpose of the added `map` calls? They seem to always replace the cell value by `firstField` and `secondField`, respectively.

Reply:
Yes, that's exactly what's happening. The reason here is that `CellFactory` is expecting the field name to choose the icon.
I thought about trying to identify if the column contains a DOI or an URL, but I think this is the better approach, as here this information is already there. 
I'm open for different approaches though.

Reply:
ok, makes sense. The code looks a bit strange but I'm not able to find a more readable solution. Could you please add a short comment explaining the construction. Thanks!

Reply:
I used `EasyBind.map()` directly, this makes this line somewhat more readable. Additionally to the comment you suggested.